qiime2/sdk/plugin_manager.py

```diff
@@ -79,20 +79,43 @@ def _integrate_plugin(self, plugin):
             self.transformers[input][output] = transformer_record
 
         for name, record in plugin.formats.items():
+
             if name in self.formats:
                 raise NameError(
                     "Duplicate format registration (%r) defined in plugins: %r"
                     " and %r" %
                     (name, record.plugin.name, self.formats[name].plugin.name)
                 )
             self.formats[name] = record
+
         self.type_formats.extend(plugin.type_formats)
 
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
     @property
+    def importable_formats(self):
+        """Return dict of formats that are importable.
+
+        A format is importable if an installed plugin can transform it into
+        one of the canonical semantic types.
+
+        """
+        _importable_formats = {}
```

Comment:
Done, and when you review this look at the `_post_init()` function to see my precise implementation as well as my slight modification of `importable_types`.